ci/build-fips-action/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM amazonlinux:2
+
+ARG TARGETARCH
+
+ENV TARGETARCH=$TARGETARCH
+
+RUN yum groupinstall -y "Development Tools" && yum install -y curl git
+
+RUN curl -Lo go.tar.gz https://go.dev/dl/go1.20.5.linux-$TARGETARCH.tar.gz
```

Comment:
Not happy about the hard-coded go version here, but I understand why it is. Going to be a bit of a pain when we update go.